examples/advanced_agents/k8s_infra_monitor/src/nat_k8s_infra_monitor/resource_pressure_tool.py

```
@@ -0,0 +1,167 @@
+    memory_threshold: int,
+) -> str:
+    """Execute kubectl commands to gather live resource utilization data."""
+    cmd_base = ["kubectl"]
+    if kubeconfig_path:
+        cmd_base.extend(["--kubeconfig", kubeconfig_path])
+
+    sections: list[str] = []
+
+    # Get node conditions (pressure indicators)
+    try:
+        result = subprocess.run(
```

Ruff (0.15.6)

[error] `subprocess` call: check for execution of untrusted input

(S603)